From: Lee Duncan <lduncan@suse.com>
Date: Thu 29 Sep 2022 09:46:47 AM PDT
Subject: [PATCH] kABI: fix adding another field to scsi_device
Patch-mainline: Never, kABI fix
References: bsc#1203039

Patch scsi-core-Add-BLIST_NO_ASK_VPD_SIZE-for-some-VDASD.patch
added field no_ask_vpd_sz_first, so hide that from the kABI
checker. There should be plenty of room for one more bit.

[lduncan: refreshed to apply]
[mkoutny: silence_suspend is part of SPE15-SP5 kABI already but we cannot hide
          the rename of no_ask_vpd_sz_first to no_vpd_size, so keep both ]
---
 include/scsi/scsi_device.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/scsi/scsi_device.h
+++ b/include/scsi/scsi_device.h
@@ -216,7 +216,11 @@ struct scsi_device {
 					 * creation time */
 	unsigned ignore_media_change:1; /* Ignore MEDIA CHANGE on resume */
 	unsigned silence_suspend:1;	/* Do not print runtime PM related messages */
+	unsigned no_ask_vpd_sz_first:1;	/* kABI: dummy member */
+#ifndef __GENKSYMS__
+	/* kABI: there should be room for this single bit, rename of no_ask_vpd_sz_first */
 	unsigned no_vpd_size:1;		/* No VPD size reported in header */
+#endif
 
 	unsigned int queue_stopped;	/* request queue is quiesced */
 	bool offline_already;		/* Device offline message logged */
